### DISCUSSION TITLE
[&>div:nth-child(n+2)]

### DISCUSSION THREAD

**Discussion by harisrahim5011** [tailwindlabs/tailwindcss#14806] [Help] [answered]
[&>div:nth-child(n+2)] is not working on my local file but working fine on tailwind css playground

**Comment by wongjn** [comment]
Consider checking:
- Tailwind has compiled since you added the class.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). 
- The class name would resolve to a Tailwind class that Tailwind would generate a rule for.

Otherwise, consider providing a project that reproduces the unexpected behavior.

**Comment by aniruddhaadak80** [comment]
Hi @harisrahim5011 ,

It seems like you're encountering an issue with the `[&>div:nth-child(n+2)]` selector working locally but not as expected. Here are a few things to check:

1. **Ensure Tailwind CSS is Properly Installed**: Make sure your local setup has Tailwind CSS properly configured. You can check your `tailwind.config.js` file for any custom configurations that might affect this selector.

2. **Check Purge Settings**: If you're using a purge setting in your Tailwind CSS configuration, it might be stripping out unused styles. Ensure that your local build process is not removing necessary styles by including your HTML files in the purge option.

3. **Inspect Build Output**: Sometimes, the build output can differ between environments. Make sure to inspect the compiled CSS in your local environment to verify that the styles are present.

4. **Browser Compatibility**: Ensure that your browser supports the feature you're using. The `:nth-child` selector is widely supported, but if you're using other advanced selectors or features, check compatibility.

Thanks!

**Comment by UtkarshK95** [comment, marked as answer]
If `[&>div:nth-child(n+2)]` works in the Tailwind CSS playground but not in your local environment, it could be due to a configuration issue or version mismatch. Here are some steps to troubleshoot and ensure it works locally:

### 1. **Verify Tailwind CSS Version**
   - The `[&>...]` syntax requires **JIT mode** (Just-In-Time mode), which is default in Tailwind CSS v3 and above. Check that you’re using the latest version of Tailwind CSS:
     ```bash
     npm list tailwindcss
     ```
   - If you’re using an older version, update it by running:
     ```bash
     npm install -D tailwindcss@latest
     ```

### 2. **Check Your `tailwind.config.js` File**
   - Ensure your `tailwind.config.js` file includes all relevant paths in the `content` section so Tailwind can detect and compile all classes correctly:
     ```js
     /** @type {import('tailwindcss').Config} */
     export default {
       content: [
         './index.html',
         './src/**/*.{js,jsx,ts,tsx,html}',
       ],
       theme: {
         extend: {},
       },
       plugins: [],
     };
     ```

### 3. **Use JIT-Compatible `@layer` Directive if Needed**
   - If you’re trying to apply this within a custom CSS file, make sure to wrap it in an `@layer` directive:
     ```css
     @layer components {
       .parent-selector {
         @apply [&>div:nth-child(n+2)]:text-blue-500;
       }
     }
     ```
   - This ensures that Tailwind recognizes the custom CSS structure and applies the utility.

### 4. **Restart the Development Server**
   - Sometimes changes in the config or CSS files aren’t applied immediately. Restart your development server to ensure Tailwind picks up the latest changes:
     ```bash
     npm run dev
     ```

### 5. **Use a Custom Class as a Fallback**
   - If `[&>div:nth-child(n+2)]` still doesn’t work, try setting a custom class and applying the desired styles as a fallback:
     ```css
     .custom-child-selector > div:nth-child(n+2) {
       @apply text-blue-500;
     }
     ```
     
After these adjustments, the `[&>div:nth-child(n+2)]` selector should work as expected in your local setup.

**Reply by harisrahim5011** [reply]
Option 5 works, custom class

**Comment by harisrahim5011** [comment]
Custom class works

**Comment by harisrahim5011** [comment]
Great thanks to all of you.
